packages/core/src/state-management/__tests__/repository.test.ts

```diff
@@ -637,3 +649,105 @@ describe('applyWriteOpts', () => {
+      const tile = await TileDocument.create(ceramic, { text: 1 }, undefined, { anchor: true })
+      await tile.update({ text: 2 }, undefined, { anchor: true })
+      const state$ = await repository.load(tile.id)
+      const event: AnchorEvent = {
+        status: AnchorRequestStatusName.COMPLETED,
+        message: 'Last COMPLETED',
+        streamId: tile.id,
+        cid: tile.tip,
+        witnessCar: generateFakeCarFile(),
+      }
+      const handleAnchorCommitSpy = jest.spyOn(repository as any, '_handleAnchorCommit')
+      // Work around verification for a fake witness car
+      handleAnchorCommitSpy.mockImplementation(async (state$: RunningState) => {
+        state$.next({
+          ...state$.value,
+          anchorStatus: AnchorStatus.ANCHORED,
+        })
+      })
+      const shouldRemove = await repository.handleAnchorEvent(state$, event)
+      expect(shouldRemove).toBeTruthy()
+      expect(handleAnchorCommitSpy).toBeCalled()
+      expect(state$.state.anchorStatus).toEqual(AnchorStatus.ANCHORED)
+    })
+
+    test('Anchor FAILED for tip should update the stream state and be removed from the anchorRequestStore', async () => {
+      const tile = await TileDocument.create(ceramic, { text: 1 }, undefined, { anchor: true })
+      await tile.update({ text: 2 }, undefined, { anchor: true })
+      const state$ = await repository.load(tile.id)
+      const event: AnchorEvent = {
+        status: AnchorRequestStatusName.FAILED,
+        message: 'Last FAILED',
+        streamId: tile.id,
+        cid: tile.tip,
+      }
+      const shouldRemove = await repository.handleAnchorEvent(state$, event)
+      expect(shouldRemove).toBeTruthy()
+      expect(state$.state.anchorStatus).toEqual(AnchorStatus.FAILED)
+    })
+  })
+
+  describe('for non-tip', () => {
+    test('Anchor COMPLETED for non tip should remove the request from the store', async () => {
+      // The anchor commit gets applied anyway
+      const tile = await TileDocument.create(ceramic, { text: 1 }, undefined, { anchor: true })
+      await tile.update({ text: 2 }, undefined, { anchor: true })
+      const state$ = await repository.load(tile.id)
+      const event: AnchorEvent = {
+        status: AnchorRequestStatusName.COMPLETED,
+        message: 'First COMPLETED',
+        streamId: tile.id,
+        cid: tile.state.log[0].cid,
+        witnessCar: generateFakeCarFile(),
+      }
+      const handleAnchorCommitSpy = jest.spyOn(repository as any, '_handleAnchorCommit')
+      handleAnchorCommitSpy.mockImplementation(async (state$: RunningState) => {
+        state$.next({
+          ...state$.value,
+          anchorStatus: AnchorStatus.ANCHORED,
+        })
+      })
+      const shouldRemove = await repository.handleAnchorEvent(state$, event)
+      expect(shouldRemove).toBeTruthy()
+      expect(handleAnchorCommitSpy).toBeCalled()
+    })
+
+    test('Anchor REPLACED for non tip should remove the request from the store', async () => {
```

Comment:
shouldn't this NOT remove it from from the store if the REPLACED status is coming for a commit that isn't the current tip?

Reply:
actually ditto for the other two tests in this "non-tip" suite.  Any changes to the state of the request for commits other than the tip shouldn't change the fact that we need to keep polling for the tip ever, right?

Reply:
"keep polling for the tip" = we do not know about the tip, so REPLACED is a terminal state ceteris paribus.